ablySpec/Utilities.swift

```diff
@@ -0,0 +1,137 @@
+                    expect(receivedFoo2).to(equal(123))
+                    expect(receivedBar).to(beNil())
+                    expect(receivedFooOrBar).to(equal(123))
+                    expect(receivedAll).to(equal(123))
+                    
+                    eventEmitter.emit("bar", with:456)
+                    
+                    expect(receivedFoo1).to(equal(123))
+                    expect(receivedFoo2).to(equal(123))
+                    expect(receivedBar).to(equal(456))
+                    expect(receivedFooOrBar).to(equal(456))
+                    expect(receivedAll).to(equal(456))
+                    
+                    eventEmitter.emit("qux", with:789)
+                    
+                    expect(receivedFooOrBar).to(equal(456))
+                    expect(receivedAll).to(equal(789))
+                }
+                
+                it("should only call once listeners once for its event") {
+                    eventEmitter.emit("foo", with: 123)
+
+                    expect(receivedBarOnce).to(beNil())
+                    expect(receivedAllOnce).to(equal(123))
+                    
+                    eventEmitter.emit("bar", with: 456)
+                    
+                    expect(receivedBarOnce).to(equal(456))
+                    expect(receivedAllOnce).to(equal(123))
+
+                    eventEmitter.emit("bar", with: 789)
+                    
+                    expect(receivedBarOnce).to(equal(456))
+                    expect(receivedAllOnce).to(equal(123))
+                }
+                
+                it("should stop receiving events when calling off") {
+                    eventEmitter.offAll(listenerFoo1!)
+                    eventEmitter.emit("foo", with: 123)
+                    
+                    expect(receivedFoo1).to(beNil())
+                    expect(receivedFoo2).to(equal(123))
+                    expect(receivedFooOrBar).to(equal(123))
+                    expect(receivedAll).to(equal(123))
+                    
+                    eventEmitter.off("foo", listener: listenerFooOrBar!)
+                    eventEmitter.emit("foo", with: 456)
+                    
+                    expect(receivedFoo2).to(equal(456))
+                    expect(receivedFooOrBar).to(equal(123))
+                    
+                    eventEmitter.emit("bar", with: 789)
+                    
+                    expect(receivedFooOrBar).to(equal(789))
+                    
+                    eventEmitter.off("foo", listener: listenerAll!)
+                    eventEmitter.emit("foo", with: 111)
+                    
+                    expect(receivedAll).to(equal(789))
+                    
+                    eventEmitter.emit("bar", with: 222)
+                    
+                    expect(receivedAll).to(equal(222))
+                }
+
+                it("should receive only once after calling onAll and then once") {
```

Comment:
This test is wrong as far as I understand.  If you register a listener twice, then it will be called twice. There is no de-duping of events by listener.  Try this Node.js example yourself:

```javascript
const EventEmitter = require('events');
const util = require('util');

function MyEmitter() {
  EventEmitter.call(this);
}
util.inherits(MyEmitter, EventEmitter);

const myEmitter = new MyEmitter();
let listener = function() {
  console.log('an event occurred!');
};
myEmitter.on('event', listener);
myEmitter.on('event', listener);
myEmitter.once('event', listener);

console.log("> Emitting event first time");
myEmitter.emit('event');

console.log("> Emitting event second time");
myEmitter.emit('event');

/* 
 
> Emitting event first time
> myEmitter.emit('event');
an event occurred!
an event occurred!
an event occurred!

> Emitting event second time
> myEmitter.emit('event');
an event occurred!
an event occurred!

*/
```